packages/fast-components-react-msft/src/pivot/pivot.tsx

```diff
@@ -0,0 +1,287 @@
+import * as React from "react";
+import { get } from "lodash-es";
+import * as ReactDOM from "react-dom";
+import { canUseDOM } from "exenv-es6";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import { canUseFocusVisible } from "@microsoft/fast-web-utilities";
+import {
+    PivotHandledProps,
+    PivotManagedClasses,
+    PivotUnhandledProps,
+} from "./pivot.props";
+import {
+    ManagedClasses,
+    PivotClassNameContract,
+} from "@microsoft/fast-components-class-name-contracts-msft";
+import { Direction } from "@microsoft/fast-application-utilities";
+import { toPx } from "@microsoft/fast-jss-utilities";
+import { TabsClassNameContract } from "@microsoft/fast-components-react-base";
+import { Tabs as BaseTabs } from "@microsoft/fast-components-react-base";
+import { PivotProps } from ".";
+
+export interface PivotState {
+    offsetX: number;
+    activeId: string;
+    focused: boolean;
+}
+
+class Pivot extends Foundation<PivotHandledProps, PivotUnhandledProps, PivotState> {
+    public static displayName: string = "Pivot";
+
+    /**
+     * React life-cycle method
+     */
+    public static getDerivedStateFromProps(
+        nextProps: PivotProps,
+        prevState: PivotState
+    ): null | Partial<PivotState> {
+        if (nextProps.activeId && nextProps.activeId !== prevState.activeId) {
+            return {
+                activeId: nextProps.activeId,
+            };
+        }
+
+        return null;
+    }
+
+    protected handledProps: HandledProps<PivotHandledProps> = {
+        label: void 0,
+        items: void 0,
+        managedClasses: void 0,
+    };
+
+    private tabsRef: React.RefObject<any>;
+
+    private ltr: Direction;
+
+    private tabPanelIndex: number;
+
+    /**
+     * The constructor
+     */
+    constructor(props: PivotProps) {
+        super(props);
+
+        const activeId: string = !this.props.activeId
+            ? this.props.items
+                ? this.props.items[0].id
+                : ""
+            : this.props.activeId;

```

Comment:
We should ensure that this is an array before treating it as one. I'd suggest taking a look at carousel - we can use get from lodash and limit how much we need to actually execute here.

Reply:
Carousel assumes items is a required prop... It is not in Base tabs. so if activeId is not set and items is not passsed it fails

Reply:
I'll add items back to pivot props and make it required.

Reply:
@eljefe223 That's not the issue here - that's a typescript assumption. I'm speaking purely in terms of JS. We should ensure that items is an array before we treat it as such.

Reply:
Agreed that we need that check. I'm just saying that I think its valid change to make items a required props here